src/main/java/seedu/address/logic/parser/NewCommandParser.java

```diff
@@ -48,9 +50,10 @@ public NewCommand parse(String args) throws ParseException {
         Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
         IngredientList ingredients = ParserUtil.parseIngredients(argMultimap.getValue(PREFIX_INGREDIENTS).get());
         InstructionList instructions = ParserUtil.parseInstructions(argMultimap.getValue(PREFIX_INSTRUCTIONS).get());
+        Calorie calorie = ParserUtil.parseCalorie(argMultimap.getValue(PREFIX_CALORIE).get());
```

Comment:
change this line to:
`Calorie calorie = ParserUtil.parseCalorie(argMultimap.getValue(PREFIX_CALORIE).orElse("0"));`
for an easy fix.

your original code calls `get()`, which expects a value to be present. but since `cal/` is optional, there may or may not be a value present, which is why it throws an exception when there's no value.